### PR TITLE
Add RTC API and driver for sifive,rtc0

### PR DESCRIPTION
**Pull request by nategraff-sifive**

### CODE REVIEW

File: metal/rtc.h [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+ * @brief List of RTC run behaviors
+ */
+enum metal_rtc_run_option {
+    METAL_RTC_STOP = 0,
+    METAL_RTC_RUN,
+};
+
+struct metal_rtc_vtable {
+    uint64_t (*get_rate)(const struct metal_rtc *const rtc);
+    uint64_t (*set_rate)(const struct metal_rtc *const rtc, const uint64_t rate);
```
Comment (bsousi5) [inline metal/rtc.h]: any special reason these need to be 64bits?

File: metal/rtc.h [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+ */
+struct metal_rtc {
+    const struct metal_rtc_vtable *vtable;
+};
+
+/*!
+ * @brief Get the rate of the RTC
+ * @return The rate in Hz
+ */
+inline uint64_t metal_rtc_get_rate(const struct metal_rtc *const rtc) {
```
Comment (bsousi5) [inline metal/rtc.h]: For 32bits system, does not make sense to store or use 54bits value. May be uintptr_ptr type be more appropriate.

File: src/drivers/sifive_rtc0.c [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    count <<= 32;
+    count |= RTC_REGW(base, METAL_SIFIVE_RTC0_RTCCOUNTLO);
+
+    return count;
+}
+
+uint64_t __metal_driver_sifive_rtc0_set_count(const struct metal_rtc *const rtc, const uint64_t count) {
+    const uint64_t base = __metal_driver_sifive_rtc0_control_base(rtc);
+
+    RTC_REGW(base, METAL_SIFIVE_RTC0_RTCCOUNTHI) = (UINT_MAX & (count >> 32));
```
Comment (bsousi5) [inline src/drivers/sifive_rtc0.c]: on low end cores, does RTC count regs are also 64bits wide?

File: metal/rtc.h (hunk 1 above)
Comment (nategraff-sifive) [reply metal/rtc.h]: The value is a clock rate, so I'm just making sure that large clock rates don't overflow. That said, this is an RTC API, so maybe it's inappropriate to support rates over 4.3 GHz.

File: src/drivers/sifive_rtc0.c (hunk 3 above)
Comment (nategraff-sifive) [reply src/drivers/sifive_rtc0.c]: Yes, it's a 64-bit register.

**Pull request merged**